Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -878,6 +878,84 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  public override func visit(_ node: PatternBindingSyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if let tuplePatternSyntax = node.pattern.as(TuplePatternSyntax.self) {
+      for element in tuplePatternSyntax.elements where element.hasError && element.trailingComma?.isMissingAllTokens == true {
+        if let previousToken = node.previousToken(viewMode: .sourceAccurate) {
+          var fixIts: [FixIt] = []
+
+          if let identifierPatternSyntax = element.pattern.as(IdentifierPatternSyntax.self), let nextIdentifier = element.nextToken(viewMode: .sourceAccurate)?.as(TokenSyntax.self), nextIdentifier.rawTokenKind == .identifier {
+            let tokens = [identifierPatternSyntax.identifier, nextIdentifier]
+
+            let joined = previousToken.text + tokens.map(\.text).joined()
+            fixIts = [
+              FixIt(
+                message: .joinIdentifiers,
+                changes: [
+                  [.replace(oldNode: Syntax(previousToken), newNode: Syntax(TokenSyntax(.identifier(joined), presence: .present)))],
+                  .makeMissing(tokens),
+                ]
+              )
+            ]
+            if tokens.contains(where: { $0.text.first?.isUppercase == false }) {
+              let joinedUsingCamelCase = previousToken.text + tokens.map({ $0.text.withFirstLetterUppercased() }).joined()
+              fixIts.append(
+                FixIt(
+                  message: .joinIdentifiersWithCamelCase,
+                  changes: [
+                    [.replace(oldNode: Syntax(previousToken), newNode: Syntax(TokenSyntax(.identifier(joinedUsingCamelCase), presence: .present)))],
+                    .makeMissing(tokens),
+                  ]
+                )
+              )
+            }
+          }
```

Comment:
What is the common pattern? Have multiple fix its or just a single? 

Because if you type `fooBar` like in `TypeTests.testMissingColonInType` then a wrong space is as common error as a missing `:` IMO.

Reply:
That’s a very good question. Our infrastructure does support multiple Fix-Its, as with the joining of identifiers but the tricky thing is that we need to pick one “primary” fix of how we want to represent the incorrect code in the fixed up syntax tree. So, in most cases I would expect there to be a single Fix-it.

An idea I have discussed with @flashspys in the context of https://github.com/apple/swift-syntax/pull/886 (but IIRC there’s no PR for that) is that we could check if the next identifier starts with a lowercase or uppercase letter. If it starts with a lowercase letter, then it’s likely a variable name and you want to insert a comma and if it starts with an uppercase letter it’s likely a type name and you’re likely missing a colon.

Reply:
Added your suggestion. It sounds reasonable for me, that we suggest adding a comma for `foo bar` and removing the space for `foo Bar`